pkg/anonymization/anonymizer_test.go

```diff
@@ -0,0 +1,201 @@
+		{
+			originalIP: net.IPv4(127, 0, 0, 0),
+			nextIP:     net.IPv4(127, 0, 0, 1),
+			mask:       net.IPv4Mask(255, 255, 255, 0),
+			overflow:   false,
+		},
+		{
+			originalIP: net.IPv4(192, 168, 0, 1),
+			nextIP:     net.IPv4(192, 168, 0, 2),
+			mask:       net.IPv4Mask(255, 255, 0, 0),
+			overflow:   false,
+		},
+		{
+			originalIP: net.IPv4(192, 168, 0, 254),
+			nextIP:     net.IPv4(192, 168, 0, 255),
+			mask:       net.IPv4Mask(255, 255, 0, 0),
+			overflow:   false,
+		},
+		{
+			originalIP: net.IPv4(192, 168, 0, 255),
+			nextIP:     net.IPv4(192, 168, 1, 0),
+			mask:       net.IPv4Mask(255, 255, 0, 0),
+			overflow:   false,
+		},
+		{
+			originalIP: net.IPv4(192, 168, 255, 255),
+			nextIP:     net.IPv4(192, 168, 0, 0),
+			mask:       net.IPv4Mask(255, 255, 0, 0),
+			overflow:   true,
+		},
+		{
+			originalIP: net.IPv4(10, 0, 0, 54),
+			nextIP:     net.IPv4(10, 0, 0, 55),
+			mask:       net.IPv4Mask(255, 255, 255, 254),
+			overflow:   false,
+		},
+		{
+			originalIP: net.IPv4(10, 0, 0, 55),
+			nextIP:     net.IPv4(10, 0, 0, 54),
+			mask:       net.IPv4Mask(255, 255, 255, 254),
+			overflow:   true,
+		},
+		{
+			originalIP: net.IPv4(255, 255, 255, 255),
+			nextIP:     net.IPv4(255, 255, 255, 255),
+			mask:       net.IPv4Mask(255, 255, 255, 255),
+			overflow:   true,
+		},
+		{
+			originalIP: net.IPv4(255, 255, 255, 255),
+			nextIP:     net.IPv4(0, 0, 0, 0),
+			mask:       net.IPv4Mask(0, 0, 0, 0),
+			overflow:   false,
+		},
+		// IPv6
+		{
+			originalIP: net.IP{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+			nextIP:     net.IP{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1},
+			mask:       net.IPMask{255, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+			overflow:   false,
+		},
+		// IPv6
+		{
+			originalIP: net.IP{16, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 255, 255, 255},
+			nextIP:     net.IP{16, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0},
+			mask:       net.IPMask{255, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+			overflow:   false,
+		},
+	}
+
+	for _, testCase := range testCases {
+		nextIP, overflow := getNextIP(testCase.originalIP, testCase.mask)
+		assert.True(
+			t,
+			nextIP.Equal(testCase.nextIP),
+			"IP %v and %v are not equal",
+			nextIP.String(),
+			testCase.nextIP,
+		)
+		assert.Equal(t, overflow, testCase.overflow)
+	}
+}
+
+type mockConfigProvider struct {
+	config *config.Controller
+}
+
+func (mcp mockConfigProvider) Config() *config.Controller {
+	return mcp.config
+}
+
+func getAnonymizer(t *testing.T) *Anonymizer {
+	clusterBaseDomain := "example.com"
+	networks := []string{
+		"127.0.0.0/8",
+		"192.168.0.0/16",
+	}
+
+	anonymizer, err := NewAnonymizer(clusterBaseDomain, networks)
+	assert.NoError(t, err)
+
+	return anonymizer
+}
+
+func TestAnonymizer(t *testing.T) {
+	anonymizer := getAnonymizer(t)
+
+	type testCase struct {
+		before string
+		after  string
+	}
+
+	nameTestCases := []testCase{
+		{"node1.example.com", "node1.<CLUSTER_BASE_DOMAIN>"}, // TODO:
```

Comment:
TODO ? :) More cases?

Reply:
Hm, don't remember what I meant, probably I just forgot to remove some reminder for myself :)

Reply:
OK np